src/main/java/edu/wpi/grip/ui/util/CSSElementUtility.java

```diff
@@ -0,0 +1,59 @@
+package edu.wpi.grip.ui.util;
+
+import edu.wpi.grip.core.Connection;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.Step;
+
+public final class CSSElementUtility {
+
+    private CSSElementUtility(){}
+
+
+    /**
+     * Returns the CSS id name for an operation
+     * @param operation The operation to get the class name for
+     * @return The CSS id for the operation. To use as a css selector then prepend the string with a '#'
+     */
+    public static String idNameFor(Operation operation){
+        return nameFor(operation).toString();
+    }
+
+    /**
+     * Return the CSS class name for a step
+     * @param step The step to get the class name for
+     * @return The CSS class for the step in the pipeline. To use as a css selector then prepend the string with a '.'
+     */
+    public static String classNameFor(Step step){
+        return shortNameFor(step.getOperation()).append("-step").toString();
+    }
+
+    /**
+     * Return the CSS class name for a connection.
+     * @param connection The connection to get the class name for.
+     * @return The CSS class for the connection in the pipeline. To use as a css selector then prepend the string with a '.'
+     */
+    public static String classNameFor(Connection connection){
+        return "connection-" +
+                connection.getOutputSocket().getSocketHint().getIdentifier()
+                + "-to-" +
+                connection.getInputSocket().getSocketHint().getIdentifier();
+    }
+
+    /**
+     * Creates the base name for the operation.
+     * @param operation The operation to generate the name for.
+     * @return The base of the operation's name.
+     */
+    private static StringBuilder nameFor(Operation operation){
```

Comment:
Why obfuscate this by splitting up into a bunch of different methods that pass around a `StringBuilder`?  I would just write the above methods using string concatenation.